src/download.ts

```diff
@@ -293,9 +293,9 @@ export async function getHnsUrl(
 
   const opts = { ...DEFAULT_DOWNLOAD_HNS_OPTIONS, ...this.customOptions, ...customOptions };
 
-  const query: Record<string, unknown> = {};
+  const query: { [key: string]: string | undefined } = {};
   if (opts.download) {
-    query.attachment = true;
+    query.attachment = "true";
```

Comment:
```suggestion
    query.attachment = true;
```

`addUrlQuery` should be happy to accept numbers and booleans